src/API/Google/Middleware.php

```diff
@@ -516,9 +516,17 @@ public function get_sdi_merchant_update_endpoint(): string {
 	 * @return string
 	 */
 	protected function get_sdi_endpoint(): string {
+		/**
+		 * Reason for using `urlencode` to encode the site URL:
+		 *
+		 * This plugin doesn't natively support linking a Google Merchant
+		 * Center account and the Google Shopping Data Integration API using a
+		 * merchant URL with paths. However, some merchants still use the
+		 * plugin's WP filter `woocommerce_gla_site_url` to achieve this.
+		*/
```

Comment:
It would be great to add a short explanation of why urlencode helps to solve the problem. From the tests and additional context, I understand that it prevents / in the value, which seems to be causing trouble.